packages/service-clients/odsp-client/src/token.ts

```diff
@@ -0,0 +1,42 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TokenResponse } from "@fluidframework/odsp-driver-definitions";
+
+/**
+ * Abstracts the token fetching mechanism for a hosting application.
+ * The hosting application is responsible for providing an implementation.
+ *
+ * @alpha
+ */
+export interface IOdspTokenProvider {
+	/**
+	 * Fetches the orderer token from host.
+	 *
+	 * @param siteUrl - Site url representing ODSP resource location. It points to the specific SharePoint site where you can store and access the containers you create.
+	 * @param refresh - Optional flag indicating whether token fetch must bypass local cache.
+	 * This likely indicates that some previous request failed authorization due to an expired token,
+	 * and so a fresh token is required.
+	 *
+	 * Default: `false`.
+	 *
+	 * NOTE: This parameter will be made required in the future.
+	 */
+	fetchWebsocketToken(siteUrl: string, refresh?: boolean): Promise<TokenResponse>;
```

Comment:
Isn't odsp-client still at a point where we can make it required _now_? Same for `fetchStorageToken`. Also, in general interfaces probably don't want to document defaults, especially interfaces we're expecting the users to implement.